papertty/drivers/drivers_full.py

```diff
@@ -270,6 +270,177 @@ def sleep(self):
+            return -1
+        # EPD hardware init start
+        self.reset()
+
+
+        self.send_command(0x12)
+        self.delay_ms(300)
+        
+        self.send_command(0x46)
+        self.send_data(0xF7)
+        self.ReadBusy()
+        self.send_command(0x47)
+        self.send_data(0xF7)
+        self.ReadBusy()
+        
+        self.send_command(0x01) # setting gaet number
+        self.send_data(0xDF)
+        self.send_data(0x01)
+        self.send_data(0x00)
+
+        self.send_command(0x03) # set gate voltage
+        self.send_data(0x00)
+
+        self.send_command(0x04) # set source voltage
+        self.send_data(0x41)
+        self.send_data(0xA8)
+        self.send_data(0x32)
+
+        self.send_command(0x11) # set data entry sequence
+        self.send_data(0x03)
+
+        self.send_command(0x3C) # set border 
+        self.send_data(0x00)
+        
+        self.send_command(0x0C) # set booster strength
+        self.send_data(0xAE)
+        self.send_data(0xC7)
+        self.send_data(0xC3)
+        self.send_data(0xC0)
+        self.send_data(0xC0)
+
+        self.send_command(0x18) # set internal sensor on
+        self.send_data(0x80)
+         
+        self.send_command(0x2C) # set vcom value
+        self.send_data(0x44)
+        
+        self.send_command(0x37) # set display option, these setting turn on previous function
+        self.send_data(0x00)     #can switch 1 gray or 4 gray
+        self.send_data(0xFF)
+        self.send_data(0xFF)
+        self.send_data(0xFF)
+        self.send_data(0xFF)  
+        self.send_data(0x4F)
+        self.send_data(0xFF)
+        self.send_data(0xFF)
+        self.send_data(0xFF)
+        self.send_data(0xFF)
+
+        self.send_command(0x44) # setting X direction start/end position of RAM
+        self.send_data(0x00)
+        self.send_data(0x00)
+        self.send_data(0x17)
+        self.send_data(0x01)
+
+        self.send_command(0x45) # setting Y direction start/end position of RAM
+        self.send_data(0x00)
+        self.send_data(0x00)
+        self.send_data(0xDF)
+        self.send_data(0x01)
+
+        self.send_command(0x22) # Display Update Control 2
+        self.send_data(0xCF)
+        self.load_lut(self.lut_1Gray_A2)
+        return 0
+
+    def load_lut(self, lut):
+        self.send_command(0x32)
+        for i in range(0, 105):
+            self.send_data(lut[i])
+
+    def display_frame(self, frame_buffer, *args):
+        if frame_buffer:
+            self.send_command(0x4E)
+            self.send_data(0x00)
+            self.send_data(0x00)
+            self.send_command(0x4F)
+            self.send_data(0x00)
+            self.send_data(0x00)
+
+            self.send_command(0x24)
+            for j in range(0, self.height):
+                for i in range(0, int(self.width / 8)):
+                    self.send_data(frame_buffer[i + j * int(self.width / 8)])   
+
+#            self.load_lut(self.lut_1Gray_A2)
```

Comment:
I guess this comment can be removed.

```suggestion
```

Reply:
It was supposed to document the original call order.